Database 1.6.0 Reference (M4) (#406)
mulesoft/docs-connectors: modules/ROOT/pages/db/database-documentation.adoc

** SECONDS
** MINUTES
** HOURS
** DAYS |  A time unit that qualifies the *Query Timeout* field |  SECONDS | 
| Fetch Size a| Number |  Indicates how many rows to fetch from the database when rows are read from a resultSet. This property is required when streaming is true; in that case a default value (10) is used. |  | 
| Max Rows a| Number |  Sets the limit for the maximum number of rows that any ResultSet object generated by this message processor can contain for the given number. If the limit is exceeded, the excess rows are silently dropped. |  | 
| SQL Query Text a| String |  The text of the SQL query to execute |  | x

> | Parameter Types a| Array of <<ParameterType>> |  Allows to optionally specify the type of one or more of the parameters in the query. If provided, you're not even required to reference all of the parameters, but you cannot reference a parameter not present in the input values |  | 

Reviewer: ```suggestion
| Parameter Types a| Array of <<ParameterType>> |  Allows to optionally specify the type of one or more of the parameters in the query. If provided, you're not required to reference all of the parameters, but you cannot reference a parameter not present in the input values. |  | 
```

** SECONDS
** MINUTES
** HOURS
** DAYS |  A time unit that qualifies the *Query Timeout* field |  SECONDS | 
| Fetch Size a| Number |  Indicates how many rows to fetch from the database when rows are read from a resultSet. This property is required when streaming is true; in that case a default value (10) is used. |  | 
| Max Rows a| Number |  Sets the limit for the maximum number of rows that any ResultSet object generated by this message processor can contain for the given number. If the limit is exceeded, the excess rows are silently dropped. |  | 
| SQL Query Text a| String |  The text of the SQL query to execute |  | x

> | Parameter Types a| Array of <<ParameterType>> |  Allows to optionally specify the type of one or more of the parameters in the query. If provided, you're not required to reference all of the parameters, but you cannot reference a parameter not present in the input values. |  | 

Reviewer: ```suggestion
| Parameter Types a| Array of <<ParameterType>> |  Allows you to optionally specify the type of one or more of the parameters in the query. If provided, you're not required to reference all of the parameters, but you cannot reference a parameter not present in the input values. |  | 
```

** SECONDS
** MINUTES
** HOURS
** DAYS |  A time unit that qualifies the *Query Timeout* field |  SECONDS | 
| Fetch Size a| Number |  Indicates how many rows to fetch from the database when rows are read from a resultSet. This property is required when streaming is true; in that case a default value (10) is used. |  | 
| Max Rows a| Number |  Sets the limit for the maximum number of rows that any ResultSet object generated by this message processor can contain for the given number. If the limit is exceeded, the excess rows are silently dropped. |  | 
| SQL Query Text a| String |  The text of the SQL query to execute |  | x

> | Parameter Types a| Array of <<ParameterType>> |  Allows to optionally specify the type of one or more of the parameters in the query. If provided, you're not even required to reference all of the parameters, but you cannot reference a parameter not present in the input values |  | 

Reviewer: The word "even" is unnecessary
Add a period to the end of the sentence

> | Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a semicolon prefix (for example, where `ID = :myParamName`).  The map's values contain the actual assignment for each parameter. |  | 

Reviewer: "semicolon" is wrong and I fixed it last time I updated this reference (which was recently). It's a colon, not a semicolon. This was incorrect in multiple places. I did a S&R. Please change back to the correct colon.

Author: Fixed throughout

** SECONDS
** MINUTES
** HOURS
** DAYS |  A time unit that qualifies the *Query Timeout* field |  SECONDS | 
| Fetch Size a| Number |  Indicates how many rows to fetch from the database when rows are read from a resultSet. This property is required when streaming is true; in that case a default value (10) is used. |  | 
| Max Rows a| Number |  Sets the limit for the maximum number of rows that any ResultSet object generated by this message processor can contain for the given number. If the limit is exceeded, the excess rows are silently dropped. |  | 
| SQL Query Text a| String |  The text of the SQL query to execute |  | x
| Parameter Types a| Array of <<ParameterType>> |  Allows to optionally specify the type of one or more of the parameters in the query. If provided, you're not even required to reference all of the parameters, but you cannot reference a parameter not present in the input values |  | 

> | Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a semicolon prefix (for example, where `ID = :myParamName`).  The map's values contains the actual assignment for each parameter. |  | 

Reviewer: ```suggestion
| Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a colon prefix (for example, where `ID = :myParamName`).  The map's values contains the actual assignment for each parameter. |  | 
```

** SECONDS
** MINUTES
** HOURS
** DAYS |  A time unit that qualifies the *Query Timeout* field |  SECONDS | 
| Fetch Size a| Number |  Indicates how many rows to fetch from the database when rows are read from a resultSet. This property is required when streaming is true; in that case a default value (10) is used. |  | 
| Max Rows a| Number |  Sets the limit for the maximum number of rows that any ResultSet object generated by this message processor can contain for the given number. If the limit is exceeded, the excess rows are silently dropped. |  | 
| SQL Query Text a| String |  The text of the SQL query to execute |  | x
| Parameter Types a| Array of <<ParameterType>> |  Allows to optionally specify the type of one or more of the parameters in the query. If provided, you're not even required to reference all of the parameters, but you cannot reference a parameter not present in the input values |  | 

> | Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a semicolon prefix (for example, where `ID = :myParamName`).  The map's values contain the actual assignment for each parameter. |  | 

Reviewer: ```suggestion
| Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a colon prefix (for example, where `ID = :myParamName`).  The map's values contain the actual assignment for each parameter. |  | 
```

** SECONDS
** MINUTES
** HOURS
** DAYS |  A time unit that qualifies the *Query Timeout* field |  SECONDS | 
| Fetch Size a| Number |  Indicates how many rows to fetch from the database when rows are read from a resultSet. This property is required when streaming is true; in that case a default value (10) is used. |  | 
| Max Rows a| Number |  Sets the limit for the maximum number of rows that any ResultSet object generated by this message processor can contain for the given number. If the limit is exceeded, the excess rows are silently dropped. |  | 
| SQL Query Text a| String |  The text of the SQL query to execute |  | x
| Parameter Types a| Array of <<ParameterType>> |  Allows to optionally specify the type of one or more of the parameters in the query. If provided, you're not even required to reference all of the parameters, but you cannot reference a parameter not present in the input values |  | 

> | Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a semicolon prefix (for example, where `ID = :myParamName`).  The map's values contain the actual assignment for each parameter. |  | 

Reviewer: ```suggestion
| Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a colon prefix (for example, where `ID = :myParamName`).  The map's values contain the actual assignment for each parameter. |  | 
```

| Input - Output Parameters a| Object |  A Map which keys are the name of a parameter to be set on the JDBC prepared statement which is both input and output.  Each parameter should be referenced in the SQL text using a semicolon prefix (for example, where `ID = :myParamName`).  The map's values will contain the actual assignment for each parameter. |  | 

> | Output Parameters a| Array of <<OutputParameter>> |  A list of output parameters to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a semicolon prefix (for example, call `multiply(:value, :result)`) |  | 

Reviewer: ```suggestion
| Output Parameters a| Array of <<OutputParameter>> |  A list of output parameters to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a colon prefix (for example, call `multiply(:value, :result)`) |  | 
```

** SECONDS
** MINUTES
** HOURS
** DAYS |  A time unit that qualifies the *Query Timeout* field |  SECONDS | 
| Fetch Size a| Number |  Indicates how many rows to fetch from the database when rows are read from a resultSet. This property is required when streaming is true; in that case a default value (10) is used. |  | 
| Max Rows a| Number |  Sets the limit for the maximum number of rows that any ResultSet object generated by this message processor can contain for the given number. If the limit is exceeded, the excess rows are silently dropped. |  | 
| SQL Query Text a| String |  The text of the SQL query to execute |  | x
| Parameter Types a| Array of <<ParameterType>> |  Allows to optionally specify the type of one or more of the parameters in the query. If provided, you're not even required to reference all of the parameters, but you cannot reference a parameter not present in the input values |  | 

> | Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a semicolon prefix (E.g: where ID = :myParamName)).  The map's values will contain the actual assignment for each parameter. |  | 

Reviewer: ```suggestion
| Input Parameters a| Object |  A Map which keys are the name of an input parameter to be set on the JDBC prepared statement. Each parameter should be referenced in the SQL text using a colon prefix (E.g: where ID = :myParamName)).  The map's values will contain the actual assignment for each parameter. |  | 
```

* <<config>> 

==== Throws

* DB:BAD_SQL_SYNTAX
* DB:CONNECTIVITY
* DB:QUERY_EXECUTION
* DB:RETRY_EXHAUSTED

== Sources

[[listener]]
=== On Table Row

`<db:listener>`

Selects from a table at a regular interval and generates one message per each obtained row.  Optionally, watermark and ID columns can be provided. 

If a watermark column is provided, the values taken from that column are used to filter the contents of the next poll, so that only rows with a greater watermark value are returned. If an ID column is provided, this component automatically makes sure that the same row is not picked twice by concurrent polls.

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | | x
| Table a| String |  The name of the table to select from |  | x
| Watermark Column a| String |  The name of the column to use for watermark. Values taken from this column are used to filter the contents of the next poll, so that only rows with a greater watermark value are processed. |  | 

> | Id Column a| String |  The name of the column to consider as row ID. If provided, this component makes sure that the same row is not processed twice by concurrent polls. |  | 

Reviewer: ```suggestion
| Id Column a| String |  The name of the column to consider as row ID. If provided, this component ensures that the same row is not processed twice by concurrent polls. |  | 
```

| Transactional Action a| Enumeration, one of:

** ALWAYS_BEGIN
** NONE |  The type of beginning action that sources can take regarding transactions. |  NONE | 
| Transaction Type a| Enumeration, one of:

** LOCAL

> ** XA |  The type of transaction to create. Availability depends on Mule version. |  LOCAL | 

Reviewer: ```suggestion
** XA |  The type of transaction to create. Availability depends on the Mule version. |  LOCAL | 
```

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required

> | In Memory Objects a| Number | The maximum amount of instances to keep in memory. If more than that is required, then it will start to buffer the content on disk. |  | 

Reviewer: ```suggestion
| In Memory Objects a| Number | The maximum number of instances to keep in memory. If more than that is required, content on the disk is buffered. |  | 
```
